### PR TITLE
[DRUP-803] Prevent installation alongside of the `apigee_edge_apidocs…

### PR DESCRIPTION
**Pull request by Jaesin**
…` module.

### CODE REVIEW

File: apigee_api_catalog.install [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Implements hook_requirements().
+ */
+function apigee_api_catalog_requirements($phase) {
+  // Prevent installation alongside of the `apigee_edge_apidocs` module.
+  if ($phase === 'install' && \Drupal::moduleHandler()->moduleExists('apigee_edge_apidocs')) {
+    return [
+      'apigee_api_catalog_module_conflict' => [
+        'title' => t('Apigee API Catalog'),
+        'description' => t('The "Apigee Edge Apidocs" module is currently installed. It is deprecated and should be uninstalled before installing the Apigee API Catalog module.'),
```
Comment (jacine) [inline apigee_api_catalog.install]: Would be good to name this the same way the module does.

```diff
-         'description' => t('The "Apigee Edge Apidocs" module is currently installed. It is deprecated and should be uninstalled before installing the Apigee API Catalog module.'),
+         'description' => t('The "Apigee Edge API Docs" module is currently installed. It is deprecated and should be uninstalled before installing the Apigee API Catalog module.'),

```

**Review by jacine — APPROVED** [review]
I just tested this, with Apigee Edge at https://github.com/apigee/apigee-edge-drupal/commit/6cfe2669e8a4d5f9dce6abf5a4e6dcca8765f9a2 and it worked great.  

DRUP-803

@Jaesin If you can make that name change real quick, that would be good...Otherwise, I think this is good to go.

File: apigee_api_catalog.install (hunk 1 above)
Comment (Jaesin) [reply apigee_api_catalog.install]: Thanks. I pushed a fix for this.

**Comment by jacine** [comment]
I'm testing out @arshad's patch at https://github.com/apigee/apigee-devportal-kickstart-drupal/pull/114, while still on this branch (with apigee edge apidocs still) and this is what I'm seeing:

m10n kickstart demo_core_install php_rewrite=ok profile=apigee_devportal_kickstart langcode=en

**Comment by Jaesin** [comment]
Nice catch. I just pushed what I hope will fix that.

**Comment by jacine** [comment]
That fixed it!  Thanks! 👍

**Comment by Jaesin** [comment]
The scenario that hasn't' been tested at this point is trying to install the `apigee_api_catalog` module with drush 9 while the `apigee_edge_apidocs` is already installed.

**Comment by Jaesin** [comment]
I installed drush 9 and it really has no respect for `hook_requirements` as mentioned in https://github.com/apigee/apigee-edge-drupal/pull/198#discussion_r286330380. If you have the `apigee_edge_apidocs` module installed, it will still install the `apigee_api_catalog` module without complaint which will result in the following error when visiting a page on the site: 

```
The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">Symfony\Component\Routing\Exception\RouteNotFoundException</em>: Route &quot;entity.apidoc.settings&quot; does not exist.
```
Should we work on figuring out a workaround for this?

**Comment by cnovak** [comment]
I added separate issue #10 so we can accept this PR w/out dealing w/Drush.

**Review by cnovak — APPROVED** [review]

**Comment by Jaesin** [comment]
Fixes: https://github.com/apigee/apigee-api-catalog-drupal/issues/5

**Pull request merged**